components/core/src/clp/regex_utils/regex_translation_utils.cpp

```diff
@@ -48,12 +57,23 @@ class TranslatorState {
     // Getters
     [[nodiscard]] auto get_state() const -> RegexPatternState { return m_state; }
 
+    [[nodiscard]] auto get_charset_start() const -> optional<string_view::const_iterator> {
+        return m_charset_start;
+    }
+
     // Setters
     auto set_next_state(RegexPatternState const& state) -> void { m_state = state; }
 
+    auto set_charset_start(string_view::const_iterator charset_start) -> void {
+        m_charset_start = charset_start;
+    }
+
+    auto invalidate_charset_start() -> void { m_charset_start = nullopt; }
+
 private:
     // Members
     RegexPatternState m_state{RegexPatternState::Normal};
+    optional<string_view::const_iterator> m_charset_start{nullopt};
```

Comment:
```suggestion
    optional<string_view::const_iterator> m_charset_start;
```
You can rely on the default constructor